plugins/inputs/t128_graphql/response_processor.go

```diff
@@ -0,0 +1,66 @@
+package t128_graphql
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Jeffail/gabs"
+)
+
+//ProcessedResponse stores the processed fields and tags for injection into telegraf accumulator
+type ProcessedResponse struct {
+	Fields map[string]interface{}
+	Tags   map[string]string
+}
+
+// ProcessResponse takes in a query response, pulls out the desired data and stores it in a struct
+func ProcessResponse(jsonChildren []*gabs.Container, collectorName string, fields map[string]string, tags map[string]string) ([]*ProcessedResponse, error) {
+	var processedResponses []*ProcessedResponse
+
+	//TODO: support queries needed by mist agent - MON-315
+	for _, child := range jsonChildren {
+		node := child.Data().(map[string]interface{})
+		processedResponse := &ProcessedResponse{Fields: map[string]interface{}{}, Tags: map[string]string{}}
+
+		for fieldRenamed, fieldName := range fields {
+			data := node[fieldName]
+
+			if strings.Contains(fieldName, "/") {
+				//TODO: clean this up, it's the second time JSONPointer() is called on the response - MON-315
+				nestedObj, err := child.JSONPointer("/" + fieldName)
+				if err != nil {
+					return nil, fmt.Errorf("unexpected response for collector %s: field %s", collectorName, fieldName)
+				}
+				data = nestedObj.Data()
+			}
+
+			if isNil(data) {
+				return nil, fmt.Errorf("found empty data for collector %s: field %s", collectorName, fieldName)
+			}
+			processedResponse.Fields[fieldRenamed] = data
+		}
+
+		for tagRenamed, tagName := range tags {
+			data := node[tagName]
+
+			if strings.Contains(tagName, "/") {
+				nestedObj, err := child.JSONPointer("/" + tagName)
+				if err != nil {
+					return nil, fmt.Errorf("unexpected response for collector %s: tag %s", collectorName, tagName)
+				}
+				data = nestedObj.Data()
+			}
```

Comment:
Maybe not worth it but this block here and above seem redundant to me. I also think this function could be split up into processFields and processTags for better readability.

Reply:
I refactored all of `response_processor.go` in my latest changes to support the mist agent queries. That code will be shared in my next PR so I'll make sure to reduce logic duplicated on tags and fields